src/app/workspace/screen/tab.tsx

```diff
@@ -0,0 +1,139 @@
+// Copyright 2023, Command Line Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from "react";
+import * as mobxReact from "mobx-react";
+import * as mobx from "mobx";
+import { sprintf } from "sprintf-js";
+import { boundMethod } from "autobind-decorator";
+import { For } from "tsx-control-statements/components";
+import cn from "classnames";
+import dayjs from "dayjs";
+import localizedFormat from "dayjs/plugin/localizedFormat";
+import { GlobalModel, GlobalCommandRunner, Session, Screen } from "../../../model/model";
+import { renderCmdText } from "../../common/common";
+import { ReactComponent as SquareIcon } from "../../assets/icons/tab/square.svg";
+import { ReactComponent as ActionsIcon } from "../../assets/icons/tab/actions.svg";
+import { ReactComponent as AddIcon } from "../../assets/icons/add.svg";
+import * as constants from "../../appconst";
+import { Reorder } from "framer-motion";
+import { MagicLayout } from "../../magiclayout";
+
+import "../workspace.less";
+import "./tabs.less";
+
+dayjs.extend(localizedFormat);
+
+@mobxReact.observer
+class ScreenTab extends React.Component<{ screen: Screen; activeScreenId: string; index: number; onSwitchScreen }, {}> {
+    tabRefs: { [screenId: string]: React.RefObject<any> } = {};
```

Comment:
why do we need (potentially) multiple refs here?  The same ScreenTab can be used by different screenIds?

Reply:
because the ScreenTab `key` is set to screen.screenId we should get a new component anytime the screenId changes (unless drag&drop is doing something weird)

Reply:
refs are used to determine the new tab index to be passed to the backend, based on their position in the DOM. Please see the `handleDragEnd` method for reference.